admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageTreeAction.java

```diff
@@ -30,296 +29,338 @@
+        assertEquals(1, errors.size());
+    }
+
+    public void testMoveForAdminUser() throws Throwable {
+        String pageToMoveCode = "pagina_12";
+        String sisterPageCode = "pagina_11";
+        IPage pageToMove = _pageManager.getDraftPage(pageToMoveCode);
+        IPage sisterPage = _pageManager.getDraftPage(sisterPageCode);
+        assertNotNull(pageToMove);
+        assertEquals(pageToMove.getPosition(), 2);
+        assertNotNull(sisterPage);
+        assertEquals(sisterPage.getPosition(), 1);
+
+        this.initAction("/do/Page", "moveUp");
+        this.setUserOnSession("admin");
+        this.addParameter("selectedNode", pageToMoveCode);
+        String result = this.executeAction();
+
+        assertEquals(Action.SUCCESS, result);
+        Collection<String> messages = this.getAction().getActionMessages();
+        assertEquals(0, messages.size());
+
+        pageToMove = this._pageManager.getDraftPage(pageToMoveCode);
+        assertEquals(pageToMove.getPosition(), 1);
+        sisterPage = this._pageManager.getDraftPage(sisterPageCode);
+        assertEquals(sisterPage.getPosition(), 2);
+
+        this.initAction("/do/Page", "moveDown");
+        this.setUserOnSession("admin");
+        this.addParameter("selectedNode", pageToMoveCode);
+        result = this.executeAction();
+
+        assertEquals(Action.SUCCESS, result);
+        messages = this.getAction().getActionMessages();
+        assertEquals(0, messages.size());
+
+        pageToMove = _pageManager.getDraftPage(pageToMoveCode);
+        assertEquals(pageToMove.getPosition(), 2);
+        sisterPage = _pageManager.getDraftPage(sisterPageCode);
+        assertEquals(sisterPage.getPosition(), 1);
+    }
+
+    public void testMoveForCoachUser() throws Throwable {
+        String pageToMoveCode = "pagina_12";
+        this.setUserOnSession("pageManagerCoach");
+        this.initAction("/do/Page", "moveDown");
+        this.addParameter("selectedNode", pageToMoveCode);
+        String result = this.executeAction();
+        assertEquals("pageTree", result);
+        Collection<String> errors = this.getAction().getActionErrors();
+        assertEquals(1, errors.size());
+    }
+
+    public void testMovementNotAllowed() throws Throwable {
+        String pageToMoveCode = "primapagina";
+        this.setUserOnSession("admin");
+        this.initAction("/do/Page", "moveUp");
+        this.addParameter("selectedNode", pageToMoveCode);
+        String result = this.executeAction();
+        assertEquals(Action.SUCCESS, result);
+        Collection<String> errors = this.getAction().getActionErrors();
+        assertEquals(1, errors.size());
+
+        pageToMoveCode = "errorpage";
+        this.initAction("/do/Page", "moveDown");
+        this.addParameter("selectedNode", pageToMoveCode);
+        result = this.executeAction();
+        assertEquals(Action.SUCCESS, result);
+        errors = this.getAction().getActionErrors();
+        assertEquals(1, errors.size());

```

Comment:
should we not also be checking which error has been returned? this applies to all the tests in this file